src/bin/nydus-image/main.rs

```diff
@@ -1479,6 +1498,52 @@ impl Command {
         }
     }
 
+    fn get_blob_meta_writer(
+        matches: &ArgMatches,
+        conversion_type: ConversionType,
+    ) -> Result<Option<Mutex<ArtifactFileWriter>>> {
+        if conversion_type == ConversionType::EStargzIndexToRef {
+            Ok(None)
+        } else if let Some(p) = matches
+            .get_one::<PathBuf>("blob-meta")
+            .map(|b| ArtifactStorage::SingleFile(b.clone()))
+        {
+            if conversion_type == ConversionType::TarToTarfs {
+                bail!(
+                    "conversion type `{}` conflicts with `--blob-meta`",
```

Comment:
What other conversion types don't support the option? Should also treat `ConversionType::EStargzIndexToRef` like this?

Reply:
Yes, the types that cannot get the original file content at build time are not supported, TarToTarfs, EStargzIndexToRef, etc